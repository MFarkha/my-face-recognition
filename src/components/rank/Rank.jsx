import React, { useEffect, useState } from "react";

const Rank = ({firstname, entries}) => {
    const [emoji, setEmoji] = useState('');

    const fetchEmoji = (rank) => {
        fetch(`${process.env.REACT_APP_LAMBDA_URL}?rank=${rank}`, {
            method: 'GET',
            headers: {
                'Content-type': 'application/json'
            }
        })
            .then(response => response.json())
            .then(data => {
                if (data.input){
                    setEmoji(data.input);
                } else {
                    throw new Error('error fetching an emojii: ', data);
                }
            })
            .catch(err => console.log('Error fetching an emojii: ', err));
    }

    useEffect(() => {
        fetchEmoji(entries)
    }, [entries])

    return (
        <div>
            <div className="white f3">
            { firstname + ', your current rank is ...' }
            </div>
            <div className="white f1">
            { entries }
            </div>
            <div className="white f3">
            { `Rank Badge: ${emoji}` }
            </div>
        </div>
    );
}

export default Rank;