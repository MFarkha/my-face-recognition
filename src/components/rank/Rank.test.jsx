import React from 'react';
import { render, screen } from '@testing-library/react';
import Rank from './Rank';

test('renders rank container correctly', () => {
    const { container } = render(<Rank firstname="John" entries="10" />);
    expect(screen.getByText(/john/i)).toBeInTheDocument();
    expect(screen.getByText(/rank/i)).toBeInTheDocument();
    expect(container).toMatchSnapshot();
});