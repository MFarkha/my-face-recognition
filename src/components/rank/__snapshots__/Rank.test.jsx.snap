// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders rank container correctly 1`] = `
<div>
  <div>
    <div
      class="white f3"
    >
      John, your current rank is ...
    </div>
    <div
      class="white f1"
    >
      10
    </div>
  </div>
</div>
`;
